### PR TITLE
Send disconnect event on node delete

### LINKED ISSUE #34: Generate disconnect event when node is deleted.
Right now only the DeleteNode response is generated even if the node had connections. Would expect it to generate a disconnect event for every connection it had.

### PR DESCRIPTION
**Pull request by setzer22**
Fixes #34

### CODE REVIEW

**Comment by IsseW** [comment]
This solution has some issues.
 - It generates disconnect events for nodes that were not connected. 
 - It doesn't generate disconnect events for connection connected to this nodes output.

File: egui_node_graph/src/graph_impls.rs [hunk 1]
```diff
@@ -63,18 +63,20 @@ impl<NodeData, DataType, ValueType> Graph<NodeData, DataType, ValueType> {
         output_id
     }
 
-    pub fn remove_node(&mut self, node_id: NodeId) {
+    /// Returns the list of input and output ids that were disconnected.
+    pub fn remove_node(&mut self, node_id: NodeId) -> (SVec<InputId>, SVec<OutputId>) {
         self.connections
             .retain(|i, o| !(self.outputs[*o].node == node_id || self.inputs[i].node == node_id));
```
Comment (IsseW) [inline egui_node_graph/src/graph_impls.rs:69]: I think a good solution would be to use Vec::drain here. And send disconnect events for those.

Comment (setzer22) [reply egui_node_graph/src/graph_impls.rs:69]: Not sure I follow here :thinking: If we drain `self.connections` we'll clear all the connections, not just the ones for this node :thinking:.

But yes, I think iterating the connections to generate the disconnect events is a better approach :+1:

Comment (IsseW) [reply egui_node_graph/src/graph_impls.rs]: Ah I was thinking about `drain_filter`, but that's an unstable feature so nevermind that.

**Comment by IsseW** [comment]
Yes, this is working for me. Thanks a lot for working on it!

**Pull request merged**